### DISCUSSION TITLE
piggy bank can't add all amount of account

### DISCUSSION THREAD

**Discussion by zhiiwg** [firefly-iii/firefly-iii#6658] [General]
Account1 has 1140.31, the piggy bank had a goal of 5,000; while trying to add the entire amount, it could only add 1140.30.
and shows info:
```
Truncated from this point <----|[2022-11-20 17:47:37] production.ERROR: Cannot add 0.01 because canAddAmount returned false.  
[2022-11-20 17:48:59] production.ERROR: Cannot add 0.01 because canAddAmount returned false.  
[2022-11-20 18:01:08] production.ERROR: Cannot add 1140.31 because canAddAmount returned false.  
```
btw piggy bank linked to the correct account(account1).

<img width="444" alt="image" src="https://user-images.githubusercontent.com/4954007/202897045-4825eed3-d994-4cdf-9a91-f48f4487c4a1.png">
<img width="341" alt="image" src="https://user-images.githubusercontent.com/4954007/202897063-8d72d03b-2995-4e93-b143-fd56943f505e.png">

1.set another piggy bank with account2, not shows "can't add 0.01" problem.
2.no problem in demo site.
3.delete piggy bank and create new piggy bank with account1 still have problem.

No idea what's happening, but seems it's a problem only with account1? 

Debug information generated at 2022-11-20 18:07:34 Asia/Shanghai for Firefly III version **5.7.15**.

| Scope | Version |
| --- | --- |
| Firefly III | 5.7.15 |
| Firefly III API | 1.5.6 |
| Build | 298, 02-11-2022 05:00:04 CET |
| DB version | 18 (exp. 18) |
| PHP | `8.1.11` |
| Host | `Linux` |

| System info | Value |
| --- | --- |
| System TZ | Asia/Shanghai |
| Browser TZ | Asia/Shanghai |
| App environment | production |
| App debug mode | false |
| App cache driver | file |
| App logging | notice, stack |
| Display errors | Off |
| Error reporting | ALL errors |
| Interface | apache2handler |
| Default language | zh_CN |
| Default locale | equal |
| BCscale | 24 |
| DB drivers | mysql, pgsql, sqlite |
| Current driver | mysql |
| Trusted proxies (.env) | `**` |

| User info | Value |
| --- | --- |
| Login provider | eloquent  |
| User guard | web  |
| Headers | N/A, N/A  |
| Stateful domains |  |
| Session start | 2022-11-01 00:00:00 |
| Session end | 2022-11-30 23:59:59 |
| Session first | 2021-03-01 00:00:00 |
| User ID | 1 |
| User language | zh_CN |
| User locale | zh_CN |
| Attempt at "zh_CN.utf8" | 'zh_CN.utf8' |
| Attempt at "zh_CN.UTF-8" | 'zh_CN.UTF-8' |
| User agent | Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.4844.84 Safari/537.36 HBPC/12.1.1.301 |

**Comment by zhiiwg** [comment]
some info from log:

```
fireflyiii-app-1  | [2022-11-20 18:45:39] production.DEBUG: Left on account: 1140.309999999999850000000000 on 2022-11-20
fireflyiii-app-1  | [2022-11-20 18:45:39] production.DEBUG: Saved so far: 0.000000000000000000000000
fireflyiii-app-1  | [2022-11-20 18:45:39] production.DEBUG: Left to save: 5000.000000000000000000000000
fireflyiii-app-1  | [2022-11-20 18:45:39] production.DEBUG: Maximum amount: 1140.309999999999850000000000
fireflyiii-app-1  | [2022-11-20 18:45:39] production.DEBUG: Compare <= 0? 1, so false
fireflyiii-app-1  | [2022-11-20 18:45:39] production.ERROR: Cannot add 1140.31 because canAddAmount returned false.
```

**Reply by JC5** [reply]
That's cool, it looks like a floating point error but Firefly III does not use floating points (exactly for this reason).

I'll check it out!
